src/ml_board/backend/restful_api/data_access.py

```diff
@@ -103,50 +144,189 @@ def load_experiment_config(top_level_logging_path: str, grid_search_id: str, exp
+
+            generator = FileDataAccess.iterfile(requested_full_path)
+            return generator
+        else:
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
+    
+    def get_experiment_config_exec(self, grid_search_id: str, experiment_id: str, config_name: str):
+        """
+        `Fetch experiment config given the experiment ID & grid search ID from event storage.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             config_name (str): Name of Configuration file
+
+        :returns: bytes response of JSON file
+        """
+        requested_full_path = os.path.realpath(os.path.join(self.top_level_logging_path, str(grid_search_id), str(experiment_id), f"{config_name}.json"))
+        if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
+            if not os.path.isfile(requested_full_path):
+                raise InvalidPathError(
+                    f"Resource {requested_full_path} not found.")
+
+            generator = FileDataAccess.iterfile(requested_full_path)
+            return generator
+        else:
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
 
     def get_checkpoint_resource(self, grid_search_id: str, experiment_id: str, epoch: str, checkpoint_resource: CheckpointResource):
+        """
+        `Fecth checkpoint resource pickle file given the experiment ID & grid search ID from event storage.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+             checkpoint_resource (CheckpointResource) : CheckpointResource type
+
+        :returns: bytes response of pickle file
+        """
         requested_full_path = os.path.realpath(os.path.join(self.top_level_logging_path, str(grid_search_id), str(experiment_id),
                                                             str(epoch), f"{checkpoint_resource}.bin"))
         if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
             if not os.path.isfile(requested_full_path):
-                raise InvalidPathError(f"Resource {requested_full_path} not found.")
+                raise InvalidPathError(
+                    f"Resource {requested_full_path} not found.")
 
             generator = FileDataAccess.iterfile(requested_full_path)
             return generator
         else:
-            raise InvalidPathError(f"File path {requested_full_path} is not safe.")
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
+
+    def post_checkpoint_resource_exec(self, grid_search_id: str, experiment_id: str, epoch: str, checkpoint_resource: CheckpointResource, payload_pickle: bytes):
```

Comment:
the function names were changed because there were same function names with different parameters and return type making it difficult to follow sometimes.
Reverted to name given.